webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -0,0 +1,141 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Store, select } from '@ngrx/store';
+import { from, of } from 'rxjs';
+import { map, exhaustMap, filter, catchError, withLatestFrom } from 'rxjs/operators';
+import { TranslateService } from '@ngx-translate/core';
+
+import { Actions as ContactActionList, ContactsActions } from '@mm-actions/contacts';
+import { ContactViewModelGeneratorService } from '@mm-services/contact-view-model-generator.service';
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+import { ContactSummaryService } from '@mm-services/contact-summary.service';
+import { TasksForContactService } from '@mm-services/tasks-for-contact.service';
+import { TargetAggregatesService } from '@mm-services/target-aggregates.service';
+
+@Injectable()
+export class ContactsEffects {
+  private contactsActions;
+  private globalActions;
+
+  constructor(
+    private actions$: Actions,
+    private store: Store,
+    private contactViewModelGeneratorService: ContactViewModelGeneratorService,
+    private contactSummaryService: ContactSummaryService,
+    private tasksForContactService: TasksForContactService,
+    private targetAggregateService: TargetAggregatesService,
+    private translateService: TranslateService,
+  ) {
+    this.contactsActions = new ContactsActions(store);
+    this.globalActions = new GlobalActions(store);
+  }
+
+  selectContact = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(ContactActionList.selectContact),
+      filter(({ payload: { id } }) => !!id),
+      exhaustMap(({ payload: { id, silent } }) => {
+        if (!silent) {
+          this.globalActions.setLoadingShowContent(id);
+        }
+        return from(this.contactViewModelGeneratorService.getContact(id, { getChildPlaces: true, merge: false })).pipe(
+          map(model => this.contactsActions.setSelected(model)),
+          catchError(error => {
+            console.error('Error selecting contact', error);
+            return of(this.globalActions.unsetSelected());
+          }),
+        );
+      }),
+    );
+  }, { dispatch: false });
+
+  setSelected = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(ContactActionList.setSelected),
+      withLatestFrom(
+        this.store.pipe(select(Selectors.getSelectedContact))
+      ),
+      exhaustMap(([{ payload: { selected } }, selectedContact]) => {
```

Comment:
@njogz  I was adding the logic for actions bar and noticed that `selected` and `selectedContact` are confusing:  because the 2 params are contacts, right? Then why or when do we need to pick "selected" or "selectedContact" for writing logic?

I think we need names that describe better the data: newSelectedContact, oldSelectedContact or storeSelectedContact, etc...

Reply:
I think the effect is running [after the store is updated](https://github.com/ngrx/platform/issues/162) so the check for refreshing the selected (L 60)will always be true and we don't need the 2 params 🤔

Reply:
The naming is indeed confusing. Will change to more descriptive variables.